packages/devtools_app_shared/lib/src/service/dtd_manager.dart

```diff
@@ -17,26 +14,29 @@ class DTDManager {
   final ValueNotifier<DTDConnection?> _connection =
       ValueNotifier<DTDConnection?>(null);
 
+  /// Whether the [DTDManager] is connected to a running instance of the DTD.
+  bool get hasConnection => connection.value != null;
+
+  /// The URI of the current DTD connection.
+  Uri? get uri => _uri;
```

Comment:
Since we call `disconnect` at the beginning of the `connect` method, `_uri` will always be set to null before we try to connect again.

From what I saw, the DTD uri isn't exposed from `DTDConnection`.